### PR TITLE
fix(MessageManager): Allow caching option of an unspecified limit

### LINKED ISSUE #7093: `MessageManager#fetch()` does not work as documented
### Issue description

There seems to be a bug in the documentation or the code itself for [`MessageManager#fetch()`](https://discord.js.org/#/docs/main/main/class/MessageManager?scrollTo=fetch).

Both parameters are listed as optional. I proceed to only make use of the `options` parameter as I don't care about specifying a message limit, but I _do_ care about caching them. TypeScript yells that the overload is not valid:

```TypeScript
<Client>.once(<Constants>.Events.CLIENT_READY, async () => {
  const channel = <Client>.channels.resolve("channel_id") as TextChannel;

  // error TS2769: No overload matches this call.
  await channel.messages.fetch({
    cache: false
  });

  console.log(channel.messages.cache.size); // 50
});
```

Upon executing the above code sample in JavaScript, 50 messages are fetched as expected as is the default but the returned messages are cached as not expected.

Either the internal code logic and typings are not correct here, or this isn't documented correctly. Personally, I'd lean towards supporting this because, why not?

### Code sample

```typescript
import { Client, Constants, Intents, TextChannel } from "discord.js";

const client = new Client({
  intents: [
    Intents.FLAGS.GUILDS,
    Intents.FLAGS.GUILD_MESSAGES
  ]
});

client.once(Constants.Events.CLIENT_READY, async () => {
  const channel = client.channels.resolve("765676229478711366") as TextChannel;

  await channel.messages.fetch({
    cache: false
  });


  console.log(channel.messages.cache.size);
});

client.login();
```

### discord.js version

13.4.0-dev

### Node.js version

Node.js 16.13.0 & TypeScript 4.4.4

### Operating system

macOS Monterey 12.0.1

### Priority this issue should have

Low (slightly annoying)

### Which partials do you have configured?

No Partials

### Which gateway intents are you subscribing to?

GUILDS, GUILD_MESSAGES

### I have tested this issue on a development release

6e57b65

### PR DESCRIPTION
**Pull request by Jiralite**
**Please describe the changes this PR makes and why it should be merged:**
This pull request resolves #7093. This implementation is actually the same as #7734, so this should feel familiar and consistent.

I have also renamed `ChannelLogsQueryOptions` to `FetchMessagesOptions` to better reflect this type definition and for consistency amongst other type definitions.

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR includes breaking changes (methods removed or renamed, parameters moved or removed)

### CODE REVIEW

File: packages/discord.js/src/managers/MessageManager.js [hunk 1]
```diff
@@ -34,40 +34,79 @@ class MessageManager extends CachedManager {
...
-   * // Get messages and filter by user id
+   * // Fetch messages and filter by a user id
    * channel.messages.fetch()
    *   .then(messages => console.log(`${messages.filter(m => m.author.id === '84484653687267328').size} messages`))
    *   .catch(console.error);
    */
-  fetch(message, { cache = true, force = false } = {}) {
-    return typeof message === 'string' ? this._fetchId(message, cache, force) : this._fetchMany(message, cache);
+  fetch(options = {}) {
+    if (!options) return this._fetchMany();
```
Comment (vladfrangu) [inline packages/discord.js/src/managers/MessageManager.js:87]: This will never trigger

File: packages/discord.js/src/managers/MessageManager.js [hunk 2]
```diff
@@ -34,40 +34,79 @@ class MessageManager extends CachedManager {
...
    * @example
-   * // Get messages and filter by user id
+   * // Fetch messages and filter by a user id
    * channel.messages.fetch()
    *   .then(messages => console.log(`${messages.filter(m => m.author.id === '84484653687267328').size} messages`))
    *   .catch(console.error);
    */
-  fetch(message, { cache = true, force = false } = {}) {
-    return typeof message === 'string' ? this._fetchId(message, cache, force) : this._fetchMany(message, cache);
+  fetch(options = {}) {
```
Comment (vladfrangu) [inline packages/discord.js/src/managers/MessageManager.js:86]:
```suggestion
  fetch(options = { cache: true }) {
```

File: packages/discord.js/src/managers/MessageManager.js [hunk 3]
```diff
@@ -34,40 +34,79 @@ class MessageManager extends CachedManager {
...
+      if (existing && !existing.partial) return existing;
+    }
+
+    const data = await this.client.rest.get(Routes.channelMessage(this.channel.id, message));
+    return this._add(data, cache);
+  }
+
+  async _fetchMany(options = {}) {
+    const data = await this.client.rest.get(Routes.channelMessages(this.channel.id), {
+      query: new URLSearchParams(options),
```
Comment (vladfrangu) [inline packages/discord.js/src/managers/MessageManager.js:106]: If this is merged after https://github.com/discordjs/discord.js/pull/7744, we need to adapt the code

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: packages/discord.js/src/managers/MessageManager.js (hunk 2 above)
Comment (Jiralite) [reply packages/discord.js/src/managers/MessageManager.js:86]: The default here should be removed as a whole. The `_add()` methods already default this property to true and also resolves your following review.

**Comment by vladfrangu** [comment]
This needs a rebase :3

**Review by SpaceEEC — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Pull request merged**